osg_configure/configure_modules/gratia.py

```diff
@@ -618,89 +426,72 @@ def _configure_htcondor_ce_probe(self):
-                    except OSError as e:
-                        self.log("Error comparing DataFolder setting in %s (%s) and condor PER_JOB_HISTORY_DIR %s:\n%s"
-                                 % (config_location, data_folder, history_dir, e),
-                                 level=logging.ERROR)
-                        valid = False
-
-            # Per Gratia-126 DataFolder must end in / otherwise gratia won't find certinfo files
-            if not data_folder.endswith('/'):
-                self.log("DataFolder setting in %s must end in a /" % config_location,
-                         level=logging.ERROR)
-                valid = False
+        if not match:
+            return True
 
-        return valid
+        data_folder = match.group(1)
+        data_folder = data_folder.strip('" \t')
+        # Per Gratia-126 DataFolder must end in / otherwise gratia won't find certinfo files
+        if not data_folder.endswith('/'):
+            self.logger.error("DataFolder setting in %s must end in a /", config_location)
+            return False
+
+        history_dir = self._get_condor_ce_history_dir()
+        if not history_dir:
+            self.logger.error(textwrap.fill(
+                """Could not verify DataFolder correctness: unable to get PER_JOB_HISTORY_DIR
+                for the schedd. This may be caused by PER_JOB_HISTORY_DIR not being defined."""
+            ))
+            return False
+
+        # os.path.samefile will die if the paths don't exist so check that explicitly (SOFTWARE-1735)
+        if not os.path.exists(data_folder):
+            self.logger.error("DataFolder setting in %s (%s) points to a nonexistent location",
+                              config_location, data_folder)
+            return False
+        elif not os.path.exists(history_dir):
+            self.logger.error("condor-ce schedd's PER_JOB_HISTORY_DIR (%s) points to a nonexistent location", history_dir)
+            return False
+        else:
+            try:
+                if os.path.samefile(data_folder, history_dir):
+                    return True
+                else:
+                    self.logger.error("DataFolder setting in %s (%s) and condor-ce PER_JOB_HISTORY_DIR (%s) "
+                                      "do not match, these settings must match!",
+                                      config_location, data_folder, history_dir)
+                    return False
+            except OSError as e:
+                self.logger.error(
+                    "Error comparing DataFolder setting in %s (%s) and condor-ce PER_JOB_HISTORY_DIR %s:\n%s",
+                    config_location, data_folder, history_dir, e)
+                return False
 
-    def _get_condor_history_dir(self, condor_config_val_bin):
-        cmd = [condor_config_val_bin, '-schedd', 'PER_JOB_HISTORY_DIR']
+    def _get_condor_ce_history_dir(self):
+        cmd = [CONDOR_CE_CONFIG_VAL, '-subsystem', 'SCHEDD', 'PER_JOB_HISTORY_DIR']
```

Comment:
```suggestion
        cmd = [CONDOR_CE_CONFIG_VAL, 'PER_JOB_HISTORY_DIR']
```
I thought we were getting rid of this

Reply:
`-subsystem SCHEDD` makes it look at `SCHEDD.PER_JOB_HISTORY_DIR` if there is one (or use regular `PER_JOB_HISTORY_DIR` if there isn't). `-schedd` makes it query the running schedd to find out what's loaded into it.

Reply:
Oh, that's good to know! Nevermind then